daemon/daemon_unix.go

```diff
@@ -375,6 +390,13 @@ func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.
 		warnings = append(warnings, "IPv4 forwarding is disabled. Networking will not work.")
 		logrus.Warnf("IPv4 forwarding is disabled. Networking will not work")
 	}
+	if hostConfig.CgroupParent != "" && daemon.usingSystemd() {
+		// CgroupParent for systemd cgroup should be named as "xxx.slice"
+		length := len(hostConfig.CgroupParent)
+		if length <= 6 || hostConfig.CgroupParent[length-6:] != ".slice" {
```

Comment:
I don't know why systemd manage like this, I do notice that cgroup paths are not the same for different subsystems, on my fedora 23, I see:
```
$ cat /proc/self/cgroup
10:devices:/user.slice
9:hugetlb:/
8:freezer:/
7:memory:/user.slice/user-1000.slice/session-3.scope
6:net_cls,net_prio:/
5:blkio:/user.slice/user-1000.slice/session-3.scope
4:perf_event:/
3:cpuset:/
2:cpu,cpuacct:/user.slice/user-1000.slice/session-3.scope
1:name=systemd:/user.slice/user-1000.slice/session-3.scope
```
If I use `--cgroup-parent`,
```
$ docker run -ti --rm --cgroup-parent user-1000.slice ubuntu bash
root@a1fb342cac19:/# cat /proc/self/cgroup
10:devices:/user.slice
9:hugetlb:/user-1000.slice/docker-a1fb342cac19675d7fe26e6d596fcf460b9f0547c4f14d6f0c506bf2cbc03b44.scope
8:freezer:/user-1000.slice/docker-a1fb342cac19675d7fe26e6d596fcf460b9f0547c4f14d6f0c506bf2cbc03b44.scope
7:memory:/user.slice/user-1000.slice/docker-a1fb342cac19675d7fe26e6d596fcf460b9f0547c4f14d6f0c506bf2cbc03b44.scope
6:net_cls,net_prio:/user-1000.slice/docker-a1fb342cac19675d7fe26e6d596fcf460b9f0547c4f14d6f0c506bf2cbc03b44.scope
5:blkio:/user.slice/user-1000.slice/docker-a1fb342cac19675d7fe26e6d596fcf460b9f0547c4f14d6f0c506bf2cbc03b44.scope
4:perf_event:/user-1000.slice/docker-a1fb342cac19675d7fe26e6d596fcf460b9f0547c4f14d6f0c506bf2cbc03b44.scope
3:cpuset:/user-1000.slice/docker-a1fb342cac19675d7fe26e6d596fcf460b9f0547c4f14d6f0c506bf2cbc03b44.scope
2:cpu,cpuacct:/user.slice/user-1000.slice/docker-a1fb342cac19675d7fe26e6d596fcf460b9f0547c4f14d6f0c506bf2cbc03b44.scope
1:name=systemd:/user.slice/user-1000.slice/docker-a1fb342cac19675d7fe26e6d596fcf460b9f0547c4f14d6f0c506bf2cbc03b44.scope
```
So if there is a slice named `user-1000.slice`, processes join it, if not, a new slice named `user-1000.slice` is created under root cgroup. I think that makes sense. And this is what default cgroup parent works (not all subsystems have `system.slice`).